src/main/java/org/broadinstitute/hellbender/utils/codecs/sampileup/SAMPileupCodec.java

```diff
@@ -0,0 +1,166 @@
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class SAMPileupCodec extends AsciiFeatureCodec<SAMPileupFeature> {
+
+    // including also space as delimiter
+    private static final Pattern SPLIT_PATTERN = Pattern.compile("\\t|( +)");
+
+    // although the minimum number of fields is 6 (see specifications), if the coverage is 0 sometimes bases and qualities does not appear
+    private static int MINIMUM_FIELDS = 4;
+    // number of maximum fields expected
+    private static final int MAXIMUM_FIELDS = 6;
+
+    // regular expression for indels
+    private static final Pattern INDEL_REGEXP = Pattern.compile("([0-9]+).*");
+    // codec file extensions
+    public static final List<String> SAM_PILEUP_FILE_EXTENSIONS = Arrays.asList("pileup", "mpileup", "samp");
+
+    public SAMPileupCodec() {
+        super(SAMPileupFeature.class);
+    }
+
+    @Override
+    public Object readActualHeader(final LineIterator lineIterator) {
+        // No header for this format
+        return null;
+    }
+
+    /**
+     * Only files with {@link #SAM_PILEUP_FILE_EXTENSIONS} could be parsed
+     * @param path path the file to test for parsability with this codec
+     * @return {@code true} if the path has the correct file extension {@link #SAM_PILEUP_FILE_EXTENSIONS}, {@code false} otherwise
+     */
+    @Override
+    public boolean canDecode(final String path) {
+        return SAM_PILEUP_FILE_EXTENSIONS.stream().anyMatch(ext -> path.endsWith("."+ext));
+    }
+
+    public SAMPileupFeature decode(String line) {
+        // Split the line
+        final String[] tokens = SPLIT_PATTERN.split(line.trim(), -1);
+        // check the number of fields
+        if(tokens.length < MINIMUM_FIELDS || tokens.length > MAXIMUM_FIELDS) {
+            throw new CodecLineParsingException("The SAM pileup line didn't have the expected number of columns. " +
+                    "Note that this codec is only valid for single-sample pileups.");
+        }
+        // starting parsing
+        final int pos = parseInteger(tokens[1], "position");
+        final byte ref = parseBase(tokens[2], "reference");
+        final int cov = parseInteger(tokens[3], "coverage");
+        // we end parsing here if coverage is 0
+        if(cov == 0) {
+            return new SAMPileupFeature(tokens[0], pos, ref, new ArrayList<>());
+        }
+        // parse the elements
+        final List<SAMPileupElement> pileupElements = parseBasesAndQuals(tokens[4], tokens[5], ref);
+        if(cov != pileupElements.size()) {
+            throw new CodecLineParsingException("THe SAM pileup line didn't have the same number of elements as the expected coverage: " + cov);
+        }
+        return new SAMPileupFeature(tokens[0], pos, ref, pileupElements);
+
+    }
+
+    @VisibleForTesting
+    protected List<SAMPileupElement> parseBasesAndQuals(final String bases, final String qualities, final byte ref) {
+        // TODO: probably a index out of bounds exception will be thrown
```

Comment:
It is because I forgot to add the try-catch block. If the qualities have less length than the bases because the pileup is malformed, the code won't reach the check of the lengths at the end of the method. I've just added the try-catch